src/core/send.c

```diff
@@ -1150,6 +1150,30 @@ QuicSendFlush(
     }
     _Analysis_assume_(Builder.Metadata != NULL);
 
+    if (Builder.Path->EcnValidationState == ECN_VALIDATION_CAPABLE) {
+        Builder.EcnEctSet = TRUE;
+    } else if (Builder.Path->EcnValidationState == ECN_VALIDATION_TESTING) {
+        if (Builder.Path->EcnTestingEndingTimeSet) {
+            if (!CxPlatTimeAtOrBefore64(
+                    CxPlatTimeUs64(), Builder.Path->EcnTestingEndingTime)) {
```

Comment:
```suggestion
                    TimeNow, Builder.Path->EcnTestingEndingTime)) {
```
We've already paid the cost to query time above. Let's use that.